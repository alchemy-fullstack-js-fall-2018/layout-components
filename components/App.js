import Component from './Component';
import Header from './Header';
import Footer from './Footer';
import Section from './Section';

export default class App extends Component {
  render() {
    const header = new Header({
      links: ['about', 'blog', 'contact']
    });

    const footer = new Footer({
      content: 'foooooteeerrrrr'
    });

    const section = new Section({
      sectionContent: 'section'
    });

    return `
      ${header.render()}
      <h1>${this.props.message}</h1>

      ${section.render()}

      ${footer.render()}
    `;
  }
}

