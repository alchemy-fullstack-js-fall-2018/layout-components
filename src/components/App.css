body {
  background-color: salmon;
}

.main {
  font-family: sans-serif;
  font-size: 30px;
  text-align: center;
}
