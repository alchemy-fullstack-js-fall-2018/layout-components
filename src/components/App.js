import Component from './Component';
import Header from './Header';
import Section from './Section';
import Footer from './Footer';
import styles from './App.css';

export default class App extends Component {
  render() {

    const header = new Header({
      links: ['about', 'blog', 'contact']
    });

    const section1 = new Section ({
      message: 'I am section 1.'
    })

    const section2 = new Section ({
      message: 'I am section 2.'
    })

    const footer = new Footer({
      peopleToThank: ['Mom', 'Dad', 'my agent']
    })

    return `
    <h1 class="${styles.main}">${this.props.message}</h1>
    ${header.render()}
    ${section1.render()}
    ${section2.render()}
    ${footer.render()}
    `;
  }
}