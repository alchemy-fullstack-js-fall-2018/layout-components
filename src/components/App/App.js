import Component from '../Component/Component';
import Header from '../Header/Header';
import Footer from '../Footer/Footer';
import Section from '../Section/Section';

import styles from './App.css';

import { teams } from '../../data/teams';

export default class App extends Component {

  render() {

    const header = new Header();
    const footer = new Footer();
    const section = new Section({ teams });


    return `
      <main id="${styles.main}">
        ${header.render()}
        ${section.render()}
        ${footer.render()}
      </main>
    `;

  }


}
