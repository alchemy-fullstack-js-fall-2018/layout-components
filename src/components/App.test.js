import App from './App';

describe('App Component', () => {
  it('should be an app', () => {
    const app = new App({ message: 'I AM APP' });

    expect(app.render()).toMatchSnapshot();
  });
});