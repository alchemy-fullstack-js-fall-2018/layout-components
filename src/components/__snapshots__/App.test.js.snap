// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component should be an app 1`] = `
"
    <h1 class=\\"main\\">I AM APP</h1>
    
      <header class=\\"header\\">
        <nav>
          <a href=\\"/about\\">about</a> <a href=\\"/blog\\">blog</a> <a href=\\"/contact\\">contact</a>
        </nav>
      </header>
    
        <section class=\\"section\\">
          <p class=\\"p\\">I am section 1.</p>
        </section>
    
    
        <section class=\\"section\\">
          <p class=\\"p\\">I am section 2.</p>
        </section>
    
    
        <footer class=\\"footer\\">
          <p>Special thanks to: Mom, Dad, my agent</p>
        </footer>
    
    "
`;
